### PR TITLE
dns/bind: Add duck duck go

### PR DESCRIPTION
**Pull request by Northguy**
@mimugmail In order to create "clean" pull requests: created a new branch and reverted the www/nginx commit d0f64ef in this pull request. Hope this is ok like this

### CODE REVIEW

**Comment by mimugmail** [comment]
The model is now missing?

File: dns/bind/src/opnsense/mvc/app/controllers/OPNsense/Bind/forms/dnsbl.xml [hunk 1]
```diff
@@ -25,6 +25,12 @@
         <type>checkbox</type>
         <help>This will force SafeSearch.</help>
     </field>
+    <field>
+        <id>dnsbl.forcesafeduckduckgo</id>
+        <label>Enable DuckDuckGo SafeSearch</label>
+        <type>checkbox</type>
+        <help>This will force SafeSearch.</help>
+    </field>
```
Comment (fabianfrz) [inline dns/bind/src/opnsense/mvc/app/controllers/OPNsense/Bind/forms/dnsbl.xml]: @mimugmail  isn't that the model part?

**Comment by fabianfrz** [comment]
@Northguy the model is this file, which defines the data structure. Without it, your data will not be stored.

https://github.com/opnsense/plugins/blob/master/dns/bind/src/opnsense/mvc/app/models/OPNsense/Bind/Dnsbl.xml

**Comment by Northguy** [comment]
Hmm,

I think I am officially "lost" now... I did change this file and it is part of the following commit: (??)

https://github.com/opnsense/plugins/pull/1085/commits/d09599a22ae44481765a8a903a7dd8998ae3cc5c

Please bear with me and guide me on the way. I do have some basic programming knowledge, but that was 20 years ago (Borland TP7 and Delphi). I do know where to look and where to search and can 'guestimate' necessary changes to your code, but Git and OPNsense code is still confusing me sometimes. 

I am editing code on a local copy through Git desktop or the web interface. Local testing: I really need to learn how to perform that. Don't know how to run OPNsense on my desktop.

**Comment by Northguy** [comment]
@mimugmail : One additional question: shouldn't we just put one checkbox for safe browsing? Something like "_Enable safe search for applicable search engines_". It doesn't make much sense to enable safe search for Google, while not enabling safe search for Bing or DuckDuckgo (or any other search engine that is added). This will prevent creating a long list of checkboxes?

**Comment by mimugmail** [comment]
Maybe separate search engines and YouTube, but for now we'll stick with current

File: dns/bind/src/opnsense/mvc/app/models/OPNsense/Bind/Dnsbl.xml [hunk 2]
```diff
@@ -36,6 +36,10 @@
             <default>0</default>
             <Required>Y</Required>
         </forcesafegoogle>
+        <forcesafeduckduckgo type="BooleanField">
+            <default>0</default>
+            <Required>Y</Required>
+        </forcesafeduckduckgo>
```
Comment (fichtner) [inline dns/bind/src/opnsense/mvc/app/models/OPNsense/Bind/Dnsbl.xml]: Required Y needs model version number increase, see above

**Comment by mimugmail** [comment]
I already bumped model in a PR before .. since 1.4 Not yet released should be ok

**Comment by fichtner** [comment]
Merged then, thanks!

**Comment by Northguy** [comment]
> Now it looks good, thanks! I installed your branch and tested successfully:

>git clone https://github.com/Northguy/plugins
> cd plugins
> git checkout Add_DuckDuckGo
> cd dns/bind
> make upgrade

Hi @mimugmail,

OK, so I suppose I could execute the above as root on my OPNsense box, to test my branch of the OPNsense master? The first 4 lines are clear to me. They will download all files into my OPNsense installation in the root of the folder where I run the commands. 

But how about the last line. The _make upgrade_ will then upgrade my installation of OPNsense with the adapted code? How do I revert changes if something is not working as expected? I am a bit anxious of testing on a running installation, without having the knowledge to revert to a working install in the unfortunate case I 'break' something.

**Comment by fichtner** [comment]
"make upgrade" will take your code in your plugin directory (dns/bind), package it and install it on the system. To revert, check out the master branch, revert your manual changes with git or manually and run "make upgrade" again. Or you can use the "opnsense-revert" tool to revert to the online repository state:

    # opnsense-revert os-bind-devel

"-devel" or no devel depends on what you previously installed, but normally development happens in the master branch or a branched branch from master so packages will receive a "-devel" suffix.

**Pull request merged**